File: leetcode/560/SubarraySum.rb
def subarray_sum(nums, k)
    #     count = 0;
    #     sum = 0;
    #     hash = {};
        
    #     for i in (0..nums.length-1)
        
    #         sum+=nums[i];
            
    #         if(sum===k)
    #             count = count +1;
    #         end
            
    #         if(hash[sum-k] !=nil)
    #             count += hash[sum-k];     
    #         end
            
    #         if(hash[sum] == nil)      
    #            hash[sum] = 1;     
    #         else
    #            hash[sum] = hash[sum] +1;         
    #         end
    #     end    
    #     return count;
          hash = Hash.new(0)
          hash[0] = 1
          count = sum = 0
          nums.each do |num|
            sum += num
            count += hash[sum - k]
            hash[sum] += 1
          end
          count
    end

File: leetcode/560/SubarraySum.ts
function subarraySum(nums: number[], k: number): number {
    let count: number = 0;
    let sum: number  = 0;
    let hash = {};
    
    for(let i:number = 0; i < nums.length; i++)
    {
        sum+=nums[i];
        
        
        if(sum===k)
            count++;
        
        if(hash[sum-k] !==undefined)
        {
            count += hash[sum-k];     
        }
        
        if(hash[sum] === undefined)
        {
           hash[sum] = 1;     
        }else
        {
           hash[sum]++;         
        }
        
    }
    return count;
};